nexus/test-utils/src/http_testing.rs

```diff
@@ -89,6 +89,7 @@ impl<'a> RequestBuilder<'a> {
             body: dropshot::Body::empty(),
             expected_status: None,
             allowed_headers: Some(vec![
+                http::header::CONNECTION,
```

Comment:
This curious change appears to be required due to [this change](https://github.com/hyperium/hyper/pull/3725) in hyper v1.5.  Without this fix, I found an update test failed, reporting that it expected a 500.  In fact it did get a 500, but it failed the validation we do on all client responses that only these allowed headers were present.  It failed that validation because there was a `connection` header present.

Reply:
interesting